core/src/test/scala/org/apache/spark/FileSuite.scala

```diff
@@ -700,4 +701,38 @@ class FileSuite extends SparkFunSuite with LocalSparkContext {
 
     assert(collectRDDAndDeleteFileBeforeCompute(true).isEmpty)
   }
+
+  test("SPARK-25100: Using KryoSerializer and" +
+      "setting registrationRequired true can lead job failed") {
+    val tempDir = Utils.createTempDir()
+    val inputFile = tempDir.getAbsolutePath + "/input"
```

Comment:
I'd rewrite this as 

```
val inputFile = new File(tempDir, "input").getAbsolutePath
```

to avoid dealing with path separator - applies on below lines as well.